docs: update `middleware.mdx` doc about revalidation issue (#870)
amannn/next-intl: docs/pages/docs/routing/middleware.mdx

  APIs](/docs/routing/navigation#localized-pathnames) in your components.
</Callout>

<details>
<summary>How to revalidate localized pathnames?</summary>

> Depending on how the route is generated, whether statically or dynamically, if you're attempting to use Next.js's function `revalidatePath()`, you will need to call it with two different pathnames.

Reviewer: ```suggestion
Depending on if a route is generated statically (at build time) or dynamically (at runtime), [`revalidatePath`](https://nextjs.org/docs/app/api-reference/functions/revalidatePath) needs to be called either for the localized or the internal pathname.
```

Let's take an example with this folder structure:
```

> .

Reviewer: ```suggestion
app
```

└── [locale]/
    └── news/
        └── [slug]
```

> And, this localized configuration :

Reviewer: ```suggestion
… and this middleware configuration:

```

```tsx filename="middleware.ts"
import createMiddleware from 'next-intl/middleware';
 
export default createMiddleware({
  defaultLocale: 'en',
  locales: ['en', 'fr'],
  pathnames: {
    '/news/[slug]': {
      en: '/news/[slug]',
      fr: '/infos/[slug]'

>     },

Reviewer: ```suggestion
    }
```

  }
});
```

> Assuming the pathname is `/fr/infos/some-article`, if the page is dynamically generated, you will need to revalidate like this: `revalidatePath('/fr/infos/some-article')`. However, if the page is statically rendered, the path to revalidate is this one: `revalidatePath('/fr/news/some-article')`. So, if you don't know if the page is statically or dynamically generated, we advise you to revalidate both.

Reviewer: Good point to revalidate both when in doubt 👍 

```suggestion
Depending on whether `some-article` was included in [`generateStaticParams`](https://nextjs.org/docs/app/api-reference/functions/generate-static-params) or not, you can invalidate the route like this:

```tsx
// Statically generated at build time
revalidatePath('/fr/news/some-article')

// Dynamically generated at runtime:
revalidatePath('/fr/infos/some-article')
```-

When in doubt, you can revalidate both paths to be on the safe side.
```

> You can find more information on the issue [here](https://github.com/amannn/next-intl/issues/846).

Reviewer: Maybe we can link to the Next.js issue? I'd like to close the issue on the `next-intl` side once your PR here is merged.

```suggestion
See also [`vercel/next.js#59825`](https://github.com/vercel/next.js/issues/59825).
```

  APIs](/docs/routing/navigation#localized-pathnames) in your components.
</Callout>

<details>

> <summary>How to revalidate localized pathnames?</summary>

Reviewer: ```suggestion
<summary>How can I revalidate localized pathnames?</summary>
```

Depending on if a route is generated statically (at build time) or dynamically (at runtime), [`revalidatePath`](https://nextjs.org/docs/app/api-reference/functions/revalidatePath) needs to be called either for the localized or the internal pathname.

Let's take an example with this folder structure:
```
app

> └── [locale]/
>     └── news/

Reviewer: ```suggestion
└── [locale]
    └── news
```

        └── [slug]
```

… and this middleware configuration:

```tsx filename="middleware.ts"
import createMiddleware from 'next-intl/middleware';
 
export default createMiddleware({
  defaultLocale: 'en',
  locales: ['en', 'fr'],
  pathnames: {
    '/news/[slug]': {
      en: '/news/[slug]',
      fr: '/infos/[slug]'
    }
  }
});
```

> Depending on whether `some-article` was included in [`generateStaticParams`](https://nextjs.org/docs/app/api-reference/functions/generate-static-params) or not, you can invalidate the route like this:

Reviewer: ```suggestion
Depending on whether `some-article` was included in [`generateStaticParams`](https://nextjs.org/docs/app/api-reference/functions/generate-static-params) or not, you can revalidate the route like this:
```
